src/factory.js

```diff
@@ -78,17 +77,17 @@ export default function plotComponentFactory(Plotly) {
             }
             throw error;
           }
-          return Plotly.newPlot(this.el, {
+          return Plotly.react(this.el, {
             data: this.props.data,
             layout: this.props.layout,
             config: this.props.config,
             frames: this.props.frames,
           });
         })
-        .then(() => this.syncWindowResize(null, true))
+        .then(() => this.syncWindowResize(shouldInvokeResizeHandler))
         .then(this.syncEventHandlers)
-        .then(this.attachUpdateEvents)
-        .then(() => this.figureCallback(this.props.onInitialized))
+        .then(() => this.figureCallback(figureCallbackFunction))
```

Comment:
why the reordering here?

Reply:
Order doesn't seem to matter - it was different between `componentWillMount` and `componentWillUpdate`. I chose one, so that I can factor it out into a separate function and call from the two lifecycle methods.